cpp/src/join/mixed_join_size_kernel.cuh

```diff
@@ -80,7 +80,7 @@ CUDF_KERNEL void __launch_bounds__(block_size)
   auto count_equality = pair_expression_equality<has_nulls>{
     evaluator, thread_intermediate_storage, swap_tables, equality_probe};
 
-  for (cudf::size_type outer_row_index = start_idx; outer_row_index < outer_num_rows;
+  for (auto outer_row_index = start_idx; outer_row_index < outer_num_rows;
        outer_row_index += stride) {
```

Comment:
Adding `stride` to `outer_row_index` when `start_index` (thread-index) is within block-size of max int32 will cause it to overflow.